inline_verifier.go

```diff
@@ -569,8 +572,46 @@ func (v *InlineVerifier) binlogEventListener(evs []DMLEvent) error {
 	}
 
 	if v.StateTracker != nil {
-		ev := evs[len(evs)-1]
-		v.StateTracker.UpdateLastResumableBinlogPositionForInlineVerifier(ev.ResumableBinlogPosition())
+		v.StateTracker.UpdateLastResumableSourceBinlogPositionForInlineVerifier(evs[len(evs)-1].ResumableBinlogPosition())
+	}
+
+	return nil
+}
+
+// Verify that all DMLs against the target are coming from Ghostferry for the
+// duration of the move. Once cutover has completed, we no longer need to perform
+// this verification as all writes from the application are directed to the target
+func (v *InlineVerifier) targetBinlogEventListener(evs []DMLEvent) error {
+	// Cutover has completed, we do not need to verify target writes
+	if v.cutoverCompleted.Get() {
+		return nil
+	}
+
+	for _, ev := range evs {
+		annotations, err := ev.Annotations()
+		if err != nil {
+			return err
+		}
+
+		foundAnnotation := false
+		for _, annotation := range annotations {
+			if strings.Contains(annotation, v.TargetDB.Marginalia) {
+				foundAnnotation = true
+				continue
+			}
+		}
+
+		if !foundAnnotation {
+			paginationKey, err := ev.PaginationKey()
+			if err != nil {
+				return err
+			}
+			return fmt.Errorf("row data with paginationKey %d on `%s`.`%s` has been corrupted", paginationKey, ev.Database(), ev.Table())
```

Comment:
"corrupted by a change directly in the target database at binlog position x, y"?

Reply:
do we want the binlog file and position instead of the pagination key? I guess we could have both?